api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -139,9 +139,11 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 							Obs replacement = os.saveObs(o, changeMessage);
 							//The logic in saveObs evicts the old obs instance, so we need to update
 							//the collection with the newly loaded and voided instance
-							toRemove.add(o);
-							toAdd.add(os.getObs(o.getId()));
-							toAdd.add(replacement);
+							if(!replacement.equals(o)) {
```

Comment:
When the session is flushed, hibernate is the one that takes care of unsetting the foreign key column for the child and when it's removed from its parent. What you're trying to ask I think was when does the parent get saved to flush changes to the collection since we don't call save, again I believe hibernate will auto flush the changes to the parent when the session is flushed